src/socks5proxy/src/socks5client.h

```diff
@@ -0,0 +1,85 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef SOCKS5CLIENT_H
+#define SOCKS5CLIENT_H
+
+#include <QObject>
+#include <QTcpSocket>
+
+class Socks5;
+
+class Socks5Client final : public QObject {
```

Comment:
The name `Socks5Client` can be a bit opaque as it could apply to both the browser trying to talk to the proxy, or the proxy being a client to some server out in the internet. I would rename this class to something like `Socks5Connection` or `Socks5Session`.